### PR TITLE
IGNITE-17801 Add support for Duration and Period types in BinaryTuple

### PR DESCRIPTION
**Pull request by ptupitsyn**
IEP updated: https://cwiki.apache.org/confluence/display/IGNITE/IEP-92%3A+Binary+Tuple+Format

### CODE REVIEW

File: modules/binary-tuple/src/main/java/org/apache/ignite/internal/binarytuple/BinaryTupleBuilder.java [hunk 1]
```diff
@@ -527,6 +529,78 @@ public BinaryTupleBuilder appendTimestamp(Instant value) {
...
+                    && Short.MIN_VALUE <= days && days <= Short.MAX_VALUE) {
+                putShort((short) years);
+                putShort((short) months);
+                putShort((short) days);
+            } else {
+                putInt(years);
+                putInt(months);
+                putInt(days);
+            }
+        }
```
Comment (ademakov) [inline modules/binary-tuple/src/main/java/org/apache/ignite/internal/binarytuple/BinaryTupleBuilder.java:589]: Isn't `1 <=  month <= 12` and `1 <= days <= 31` here? Why not to pack it in the same way as `LocalDate` ?

Comment (ptupitsyn) [reply modules/binary-tuple/src/main/java/org/apache/ignite/internal/binarytuple/BinaryTupleBuilder.java:589]: Any int value is allowed for all components, e.g. `Period.of(-100, 200, -300)`.

Comment (ademakov) [reply modules/binary-tuple/src/main/java/org/apache/ignite/internal/binarytuple/BinaryTupleBuilder.java:589]: I see, thanks.

**Review by ademakov — COMMENTED** [review]
LGTM

**Pull request merged**